routers/web/repo/view_home.go

```diff
@@ -338,14 +286,89 @@ func Home(ctx *context.Context) {
 			case ctx.Repo.TreePath != "":
 				feed.ShowFileFeed(ctx, ctx.Repo.Repository, showFeedType)
 			}
-			return
+			return true
 		}
 	}
+	return false
+}
+
+// Home render repository home page
+func Home(ctx *context.Context) {
+	if handleRepoHomeFeed(ctx) {
+		return
+	}
 
 	checkHomeCodeViewable(ctx)
 	if ctx.Written() {
 		return
 	}
 
-	renderHomeCode(ctx)
+	title := ctx.Repo.Repository.Owner.Name + "/" + ctx.Repo.Repository.Name
+	if len(ctx.Repo.Repository.Description) > 0 {
+		title += ": " + ctx.Repo.Repository.Description
+	}
+	ctx.Data["Title"] = title
+	ctx.Data["PageIsViewCode"] = true
+	ctx.Data["RepositoryUploadEnabled"] = setting.Repository.Upload.Enabled
+
+	if ctx.Repo.Commit == nil || ctx.Repo.Repository.IsEmpty || ctx.Repo.Repository.IsBroken() {
+		handleRepoEmptyOrBroken(ctx)
+		return
+	}
+
+	// Get current git entry user currently looking at.
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
+	if err != nil {
+		HandleGitError(ctx, "Repo.Commit.GetTreeEntryByPath", err)
+		return
+	}
+
+	// prepare the tree path
+	var treeNames, paths []string
+	branchLink := ctx.Repo.RepoLink + "/src/" + ctx.Repo.BranchNameSubURL()
```

Comment:
Does ctx.Repo.BranchNameSubURL() need escaping?

Reply:
No need to change, it only moves old code

Reply:
And `BranchNameSubURL` indeed does right for "URL":

![image](https://github.com/user-attachments/assets/2944fb84-97f4-4277-9bd4-abe338f819d0)